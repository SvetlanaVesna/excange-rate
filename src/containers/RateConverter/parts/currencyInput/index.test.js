import React from 'react'
import Enzyme, { shallow, mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

import CurrencyInputComponent from './index'

Enzyme.configure({ adapter: new Adapter() })

describe('<CurrencyInputComponent />', () => {
	it('snapshot test for CurrencyInputComponent component', () => {
		const props = {
			error: false,
			sourceCurrency: 'GBP',
			valueToConvert: 12,
			setValueToConvert: jest.fn(),
		}

		const component = shallow(<CurrencyInputComponent {...props} />)

		expect(component).toMatchSnapshot()
	})
	it('test for CurrencyInputComponent component if error', () => {
		const props = {
			error: true,
			sourceCurrency: 'GBP',
			valueToConvert: '',
			setValueToConvert: jest.fn(),
		}

		const component = mount(<CurrencyInputComponent {...props} />)
		const error = component.find('.input_error_label')
		expect(error.length).toEqual(1)
	})
})
