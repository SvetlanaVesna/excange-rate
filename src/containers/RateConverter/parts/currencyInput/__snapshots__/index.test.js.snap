// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CurrencyInputComponent /> snapshot test for CurrencyInputComponent component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <_default
    error={false}
    setValueToConvert={[MockFunction]}
    sourceCurrency="GBP"
    valueToConvert={12}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <span>
          <input
            className="convert_value_input"
            onChange={[Function]}
            placeholder="Input to convert"
            value={12}
          />
        </span>,
        <div>
          GBP
        </div>,
        false,
      ],
      "className": "aside aside-1",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <input
            className="convert_value_input"
            onChange={[Function]}
            placeholder="Input to convert"
            value={12}
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "convert_value_input",
            "onChange": [Function],
            "placeholder": "Input to convert",
            "value": 12,
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "GBP",
        },
        "ref": null,
        "rendered": "GBP",
        "type": "div",
      },
      false,
    ],
    "type": "aside",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span>
            <input
              className="convert_value_input"
              onChange={[Function]}
              placeholder="Input to convert"
              value={12}
            />
          </span>,
          <div>
            GBP
          </div>,
          false,
        ],
        "className": "aside aside-1",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <input
              className="convert_value_input"
              onChange={[Function]}
              placeholder="Input to convert"
              value={12}
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "convert_value_input",
              "onChange": [Function],
              "placeholder": "Input to convert",
              "value": 12,
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "GBP",
          },
          "ref": null,
          "rendered": "GBP",
          "type": "div",
        },
        false,
      ],
      "type": "aside",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
